sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,228 @@
+    assert (A.x + A.y).express(N).simplify() == sqrt(2)*N.x
+    assert A.dcm(N).simplify() == Matrix([[sqrt(2)/2, -sqrt(2)*cos(theta)/2, -sqrt(2)*sin(theta)/2], \
+        [sqrt(2)/2, sqrt(2)*cos(theta)/2, sqrt(2)*sin(theta)/2], [0, -sin(theta), cos(theta)]])
+    assert A.ang_vel_in(N) == omega*N.x
+    assert A.ang_vel_in(N).express(A).simplify() == (omega*A.x + omega*A.y)/sqrt(2)
+    assert A.ang_vel_in(N).magnitude() == omega
+    assert expand_mul(A.ang_vel_in(N).angle_between(A.x + A.y)) == 0
+    assert C.masscenter.vel(N).simplify() == (omega * A.z)/sqrt(2)
+    assert C.masscenter.pos_from(P.masscenter) == N.x - A.x
+    assert C.masscenter.pos_from(P.masscenter).express(N).simplify() == \
+        (1 - sqrt(2)/2)*N.x + sqrt(2)*cos(theta)/2*N.y + sqrt(2)*sin(theta)/2*N.z
+    assert C.masscenter.vel(N).express(N).simplify() == - sqrt(2)*omega*sin(theta)/2*N.y + sqrt(2)*omega*cos(theta)/2*N.z
+    assert C.masscenter.vel(N).angle_between(A.x) == pi/2
+
+    N, A, P, C = generate_body()
+    PinJoint('J', P, C, parent_joint_pos=N.x, child_joint_pos=A.x, child_axis=A.x+A.y-A.z)
+    assert expand_mul(N.x.angle_between(A.x + A.y - A.z)) == 0 #Axis are aligned
+    assert (A.x + A.y - A.z).express(N).simplify() == sqrt(3)*N.x
+    assert A.dcm(N).simplify() == Matrix([[sqrt(3)/3, -sqrt(6)*sin(theta + pi/4)/3, sqrt(6)*cos(theta + pi/4)/3], \
+        [sqrt(3)/3, sqrt(6)*cos(theta + pi/12)/3, sqrt(6)*sin(theta + pi/12)/3], \
+        [-sqrt(3)/3, sqrt(6)*cos(theta + 5*pi/12)/3, sqrt(6)*sin(theta + 5*pi/12)/3]])
+    assert A.ang_vel_in(N) == omega*N.x
+    assert A.ang_vel_in(N).express(A).simplify() == (omega*A.x + omega*A.y - omega*A.z)/sqrt(3)
+    assert A.ang_vel_in(N).magnitude() == omega
+    assert expand_mul(A.ang_vel_in(N).angle_between(A.x + A.y-A.z)) == 0
+    assert C.masscenter.vel(N).simplify() == (omega * A.y + omega * A.z)/sqrt(3)
+    assert C.masscenter.pos_from(P.masscenter) == N.x - A.x
+    assert C.masscenter.pos_from(P.masscenter).express(N).simplify() == \
+        (1 - sqrt(3)/3)*N.x + sqrt(6)*sin(theta + pi/4)/3*N.y - sqrt(6)*cos(theta + pi/4)/3*N.z
+    assert C.masscenter.vel(N).express(N).simplify() == sqrt(6)*omega*cos(theta + pi/4)/3*N.y + sqrt(6)*omega*sin(theta + pi/4)/3*N.z
+    assert C.masscenter.vel(N).angle_between(A.x) == pi/2
+
+    N, A, P, C = generate_body()
+    m, n = symbols('m n', positive=True)
+    PinJoint('J', P, C, parent_joint_pos=m*N.x, child_joint_pos=n*A.x, child_axis=A.x+A.y-A.z, parent_axis=N.x-N.y+N.z)
+    assert expand_mul((N.x-N.y+N.z).angle_between(A.x+A.y-A.z)) == 0 #Axis are aligned
+    assert (A.x-A.y+A.z).express(N).simplify() == \
+        (-4*cos(theta)/3 - 1/3)*N.x + (1/3 - 4*sin(theta + pi/6)/3)*N.y + (4*cos(theta + pi/3)/3 - 1/3)*N.z
+    assert A.ang_vel_in(N) == omega*N.x - omega*N.y + omega*N.z
+    assert A.ang_vel_in(N).express(A).simplify() == omega*A.x + omega*A.y - omega*A.z
+    assert A.ang_vel_in(N).magnitude() == sqrt(3)*omega
```

Comment:
The magnitude of the angular velocity of a pin join should always simply be omega, no?

Reply:
Or -omega (only different when viewing from the two frames).

Reply:
In this case there's omega in every axis(x, y, and z) , which would give `sqrt(3) * omega`, since magnitude is root of,  sum of squares of each axis' value.

Reply:
That is not correct. The angular velocity of one body wrt the other when pin jointed together is a simple angular velocity about the pin joint axis. It is always simply omega.

Reply:
> simple angular velocity about the pin joint axis. 

But parent axis is a unit in vector in every axis of parent reference frame. 

```
omega*parent axis
=> omega*(N.x - N.y + N.z) = omega*N.x - omega*N.y + omega*N.z
```

Isn't it what's supposed to be happen?

Reply:
No, that isn't what is supposed to happen. In my lectures I discuss simple rotations. The `orient_axis()` function implements a simple rotation. This rotation defines the angular velocity as omega*rotation_axis_vector. When viewing the rotation axis vector from both reference frames, it should not move. It is fixed in both frames.

Reply:
```
A.orient_axis(N, angle, N.x + N.y + N.z)
A.ang_vel(N) == angle.diff(t)
N.ang_vel(N) == -angle.diff(t)
```

Reply:
So that means if `parent_axis=N.x - N.y + N.z` then angular velocity about parent axis should be `(omega*N.x - omega*N.y + omega*N.z)/sqrt(3)` if I am not wrong.

Reply:
Here is a working example:

```
In [1]: import sympy as sm

In [2]: import sympy.physics.mechanics as me

In [3]: theta = me.dynamicsymbols('theta')

In [5]: N = me.ReferenceFrame('N')

In [6]: A = me.ReferenceFrame('A')

In [8]: A.orient_axis(N, N.x + N.y + N.z, theta)

In [11]: sm.init_printing()

In [12]: A.ang_vel_in(N)
Out[12]: 
   d                 d                 d
√3⋅──(θ(t))       √3⋅──(θ(t))       √3⋅──(θ(t))
   dt                dt                dt
─────────── n_x + ─────────── n_y + ─────────── n_z
     3                 3                 3

In [13]: A.ang_vel_in(N).magnitude()
Out[13]: 
     _____________
    ╱           2 
   ╱  ⎛d       ⎞  
  ╱   ⎜──(θ(t))⎟  
╲╱    ⎝dt      ⎠  

In [14]: N.ang_vel_in(A)
Out[14]: 
    d                   d                   d
-√3⋅──(θ(t))        -√3⋅──(θ(t))        -√3⋅──(θ(t))
    dt                  dt                  dt
───────────── n_x + ───────────── n_y + ───────────── n_z
      3                   3                   3

In [15]: N.ang_vel_in(A).magnitude()
Out[15]: 
     _____________
    ╱           2 
   ╱  ⎛d       ⎞  
  ╱   ⎜──(θ(t))⎟  
╲╱    ⎝dt      ⎠  
```

Reply:
Got it!

Reply:
This also brings up the issue of making assumptions for symbols and dynamicsymbols such that they are real.

Reply:
No effect:

```
In [16]: theta = me.dynamicsymbols('theta', real=True)

In [17]: A.orient_axis(N, N.x + N.y + N.z, theta)

In [18]: A.ang_vel_in(N).magnitude()
Out[18]: 
     _____________
    ╱           2 
   ╱  ⎛d       ⎞  
  ╱   ⎜──(θ(t))⎟  
╲╱    ⎝dt      ⎠  
```

Reply:
Yes in Joints classes I'll have taken care of assumptions, as it's new so we can correct these things here.

Reply:
But what if a person supplies there own coordinates? They may not have set the assumptions.

Reply:
I'll add that in `Notes` section to use correct assumptions.